specification/dartLangSpec.tex

```diff
@@ -1317,127 +1342,240 @@ \section{Variables}
+implicitly induces a setter with the header
+\code{\VOID\,\,\SET\,\,\id(\DYNAMIC\,\,$x$)},
+whose execution sets the value of \id{} to the incoming argument $x$.
+
+\commentary{%
+It is again assumed that type inference has taken place already,
+which could change \code{\VAR\,\,x;} to \code{$T$\,\,x;}
+and hence take us to an earlier case.%
+}
+\EndCase
+
+\LMHash{}%
+\Case{Setter: Late-final variable with declared type}
+A variable declaration of the form
+\code{\STATIC?\,\,\LATE\,\,\FINAL\,\,$T$\,\,\id;}
+implicitly induces a setter (\ref{setters}) with the header
+\code{\VOID\,\,\SET\,\,\id($T$\,\,$x$)}.
+If this setter is executed
+in a situation where the variable \id{} has not been bound,
+it will bind \id{} to the object that $x$ is bound to.
+If this setter is executed
+in a situation where the variable \id{} has been bound to an object,
+a dynamic error occurs.
+
+\commentary{%
+A late final variable declaration with no initializer is permitted.
+It introduces a variable which may be assigned to
+so long as the variable is not known to be definitely assigned.
+The property that the variable is never mutated after initialization
+is enforced dynamically rather than statically.%
+}
+\EndCase
+
+\LMHash{}%
+\Case{Setter: Late-final variable with no declared type, no initialization}
+A variable declaration of the form
+\code{\STATIC?\,\,\LATE\,\,\FINAL\,\,\id;}
+implicitly induces a setter with the header
+\code{\VOID\,\,\SET\,\,\id(\DYNAMIC\,\,$x$)}.
+An execution of said setter
+in a situation where the variable \id{} has not been bound
+will bind \id{} to the object that the argument $x$ is bound to.
+An execution of the setter
+in a situation where the variable \id{} has been bound to an object
+will incur a dynamic error.
+\EndCase
 
 \LMHash{}%
 The scope into which the implicit getters and setters are introduced
 depends on the kind of variable declaration involved.
 
 \LMHash{}%
 A library variable introduces a getter into
-the top level scope of the enclosing library.
+the library scope of the enclosing library.
 A class variable introduces a static getter into
-the immediately enclosing class.
+the body scope of the immediately enclosing class.
 An instance variable introduces an instance getter into
-the immediately enclosing class.
+the body scope of the immediately enclosing class.
 
 \LMHash{}%
-A mutable library variable introduces a setter into
-the top level scope of the enclosing library.
-A mutable class variable introduces a static setter into
-the immediately enclosing class.
-A mutable instance variable introduces an instance setter into
-the immediately enclosing class.
+A non-local variable introduces a setter if{}f it is mutable or late and final.
+A library variable which introduces a setter will introduce
+a library setter into the enclosing library scope.
+A class variable which introduces a setter will introduce 
+a static setter into the body scope of the immediately enclosing class.
```

Comment:
It would certainly be nice if we could avoid repeating this list many times, but every new defined word is also a burden on the reader who may need to look up that particular word (and just find a definition that takes 6-8 words, in which case it's faster to read those 4 words than finding that 8 word definition).

So I'm keeping the long list for now. ;-)